src/pvq_encoder.c

```diff
@@ -503,8 +504,14 @@ static int pvq_theta(od_coeff *out, const od_coeff *x0, const od_coeff *r0,
       /* PVQ search, using a gain of qcg*cg*sin(theta)*sin(qtheta) since
          that's the factor by which cos_dist is multiplied to get the
          distortion metric. */
-      cos_dist = pvq_search_rdo_double(xr, n - 1, k, y_tmp,
-       qcg*(double)cg*sin_prod*OD_CGAIN_SCALE_2, pvq_norm_lambda, prev_k);
+      if (k == 0) {
```

Comment:
Shouldn't you do this in the noref case too? Instead of duplicating code, I would move these special cases inside `pvq_search_rdo_double`